### PR TITLE
CDAP-4124 Document passing in the artifact version in the header for adding artifacts

### PR DESCRIPTION
**Pull request by awholegunch**
Add "Artifact-Versions" header. Corrected spelling in both files.

Passes [Quick Build](http://builds.cask.co/browse/CDAP-DOB57-1)

Page Of Interest: [Developers' Manual: Building Block: Plugins](http://builds.cask.co/artifact/CDAP-DOB57/shared/build-1/Docs-HTML/3.3.0-SNAPSHOT/en/developers-manual/building-blocks/plugins.html#third-party-plugins)

Fix for https://issues.cask.co/browse/CDAP-4124

### CODE REVIEW

File: cdap-docs/developers-manual/source/building-blocks/plugins.rst [hunk 1]
```diff
@@ -288,14 +293,16 @@ where ``config.json`` contains:
...
-You can deploy third-party JARs in the same way except the plugin information needs to be explicitly listed.
-As described in the documentation on :ref:`artifacts`, only snapshot artifacts can be
-re-deployed without requiring that they first be deleted.
+You can deploy third-party JARs in the same way except the plugin information needs
+:ref:`to be explicitly listed <plugins-third-party>`. As described in the documentation on
+:ref:`artifacts`, only snapshot artifacts can be re-deployed without requiring that they
+first be deleted.
 
-Using the RESTful API:
+Using the RESTful API (note that the version needs to be set explicitly, as the contents of the
```
Comment (albertshau) [inline cdap-docs/developers-manual/source/building-blocks/plugins.rst]: in a vacuum, this sounds like you always need to send the version.  Maybe qualify it? "note that if the version is not in the jar's manifest, the version needs to be set explicitly..."  Makes it sound very wordy though.

**Comment by albertshau** [comment]
one comment, otherwise lgtm

File: cdap-docs/developers-manual/source/building-blocks/plugins.rst (hunk 1 above)
Comment (awholegunch) [inline cdap-docs/developers-manual/source/building-blocks/plugins.rst]: Good catch. I've updated this in two places.

**Comment by albertshau** [comment]
:+1:

**Pull request merged**